src/main/java/com/drajer/bsa/service/impl/KarParserImpl.java

```diff
@@ -204,6 +220,18 @@ public void loadKars() {
     }
   }
 
+  private KarProcessingData makeData(HealthcareSetting hs, KnowledgeArtifact art) {
+    KarProcessingData kd = new KarProcessingData();
+    kd.setHealthcareSetting(hs);
+    kd.setKar(art);
+    kd.setEhrQueryService(ehrInterface);
```

Comment:
the last piece we also need to add here is a NotificationContext (which I know is painful to do, but that's where the app puts auth tokens)
```suggestion
    kd.setEhrQueryService(ehrInterface);
    kd.setNotificationContext(new NotificationContext());
```